packages/rbac-app/src/screens/edit.role.component.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+  Box,
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  Divider,
+  Grid,
+  Step,
+  StepLabel,
+  Stepper,
+  Typography
+} from '@mui/material'
+import {useMutation, useQuery} from '@tanstack/react-query'
+import React from 'react'
+import {useLocation, useNavigate} from 'react-router-dom'
+import Loader from '../components/helpers/loader.component'
+import {useAlert} from '../contexts/alert.context'
+import {
+  editRoleByName,
+  getApps,
+  getChannels,
+  getClients,
+  getMediators,
+  getTransactions
+} from '../services/api'
+import {Role, Routes} from '../types'
+import {AdditionalPermissionsStep} from './steps/additional.permissions.step'
+import {ChannelsClientsStep} from './steps/channels.and.clients.step'
+import {TransactionsUsersMediatorsStep} from './steps/transactions.and.users.step'
+import {useBasicBackdrop} from '../contexts/backdrop.context'
+
+const steps = [
+  'Channels & Clients',
+  'Transactions, Users & Mediators',
+  'Additional Permissions'
+]
+
+const queryFn = async () => {
+  try {
+    const [channels, clients, transactions, mediators, apps] =
+      await Promise.all([
+        getChannels(),
+        getClients(),
+        getTransactions(),
+        getMediators(),
+        getApps()
+      ])
+
+    return {channels, clients, transactions, mediators, apps}
+  } catch (err) {
+    throw err
```

Comment:
is there a way of notifying the user of an error

Reply:
yea. I'm using the react-query library. it handles the error and gives us the object

![image](https://github.com/user-attachments/assets/d7cde0a7-b1c5-4930-986f-3d812ab0370b)